### PR TITLE
Cleaner clone CLI

### PR DESCRIPTION
**Pull request by macneale4**
Fixes https://github.com/dolthub/dolt/issues/7043 for the CLI.

dolt_clone() procedure has not been touched by this PR.

### CODE REVIEW

**Comment by coffeegoddd** [comment]
@macneale4 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>99.998097 to 99.998097</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>006fbdf</TD>
<TD>not ok</TD>
<TD>113</TD>
</TR>
<TR><TD>006fbdf</TD>
<TD>ok</TD>
<TD>5937344</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>006fbdf</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>99.998097</TD>
</TR>
</table>

File: go/libraries/doltcore/env/actions/clone.go [hunk 1]
```diff
@@ -179,11 +179,20 @@ func CloneRemote(ctx context.Context, srcDB *doltdb.DoltDB, remoteName, branch s
 		return fmt.Errorf("%w; %s", ErrCloneFailed, err.Error())
 	}
 
-	branches, err := dEnv.DoltDB.GetBranches(ctx)
+	// Get all the refs from the remote. These branch remotes will be translated to remote branches, tags will
```
Comment (reltuk) [inline go/libraries/doltcore/env/actions/clone.go:182]:
```suggestion
	// Get all the refs from the remote. These branch refs will be translated to remote branch refs, tags will
```

File: go/libraries/doltcore/env/actions/clone.go [hunk 2]
```diff
@@ -179,11 +179,20 @@ func CloneRemote(ctx context.Context, srcDB *doltdb.DoltDB, remoteName, branch s
...
+	// be preserved, and all other refs will be ignored.
+	remoteRefHashes, err := dEnv.DoltDB.GetRefsWithHashes(ctx)
```
Comment (reltuk) [inline go/libraries/doltcore/env/actions/clone.go:184]: Maybe more clear is `srcRefHashes`?

**Review by reltuk — APPROVED** [review]
Looks great. A couple nit-ty comments, feel free to ignore them if you don't agree.

**Comment by coffeegoddd** [comment]
@macneale4 __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>99.998097 to 99.998097</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>5fb69eb</TD>
<TD>not ok</TD>
<TD>113</TD>
</TR>
<TR><TD>5fb69eb</TD>
<TD>ok</TD>
<TD>5937344</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>5fb69eb</TD>
<TD>5937457</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>99.998097</TD>
</TR>
</table>

**Pull request merged**